### ISSUE TITLE
Train MNIST based Linear model on CPU

### ISSUE THREAD

**Issue by nvukobratTT** [tenstorrent/tt-mlir#150]
### Summary
To have a proper e2e demo with model training on an MLIR-based compiler, we need to confirm that the model architecture is suitable to be trained on the CPU first. 

### Details
By the time of writing this issue, the linear model that is looked over for MNIST training looks like follows:
```py
class MNISTLinear(nn.Module):
    def __init__(self, input_size=784, output_size=10, hidden_size=256):
        super(MNISTLinear, self).__init__()
        self.l1 = nn.Linear(input_size, hidden_size)
        self.relu = nn.ReLU()
        self.l2 = nn.Linear(hidden_size, output_size)

    def forward(self, x):
        x = self.l1(x)
        x = self.relu(x)
        x = self.l2(x)

        return nn.functional.softmax(x)
```
Besides, the used loss is MSE, while the optimization algorithm is SGD. 

From the initial preview, this model might miss a few more layers to be able to successfully train, but that shouldn't introduce the need for any new op types. It'll be good to get accuracy with at least a **50% success rate**. 

### Bonus points (**not a requirement!**)
Train model on PyBuda/BBE stack 🪙 🪙 🪙 

### Passing criteria
- The initial (or modified) architecture of the model is trained with a 50% passing score
- If the model is modified, and new ops are introduced, followup issues for op support are required (both fwd and bwd). The same applies to loss and/or optimizer components.

**Comment by vladimirjovanovicTT** [comment]
I implemented training in PyTorch; this architecture seems to be good enough, I'm getting 95+% validation accuracy.
One detail - we should use cross-entropy instead of MSE, since this is a classification task.
I'll push it once I double-check everything.

**Comment by vladimirjovanovicTT** [comment]
Apart from this being important from the model architecture standpoint, it is good to have a fully trained model so we can load it into MLIR inference.
